test/controllers/nonsipp/sharesdisposal/SharesDisposalCYAControllerSpec.scala

```diff
@@ -0,0 +1,177 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.sharesdisposal
+
+import config.Refined.{Max50, Max5000, OneTo50, OneTo5000}
+import controllers.ControllerBaseSpec
+import controllers.nonsipp.sharesdisposal.SharesDisposalCYAController._
+import eu.timepit.refined.refineMV
+import models.HowSharesDisposed.{HowSharesDisposed, Sold}
+import models.IdentityType.Individual
+import models.SchemeHoldShare.Acquisition
+import models.SchemeId.Srn
+import models.TypeOfShares._
+import models.{CheckMode, ConditionalYesNo, IdentityType, Mode, Money, NormalMode, SchemeHoldShare, TypeOfShares}
+import org.mockito.ArgumentMatchers.any
+import pages.nonsipp.shares.{
+  CompanyNameRelatedSharesPage,
+  TypeOfSharesHeldPage,
+  WhenDidSchemeAcquireSharesPage,
+  WhyDoesSchemeHoldSharesPage
+}
+import pages.nonsipp.sharesdisposal._
+import play.api.inject.bind
+import play.api.inject.guice.GuiceableModule
+import services.PsrSubmissionService
+import uk.gov.hmrc.domain.Nino
+import views.html.CheckYourAnswersView
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class SharesDisposalCYAControllerSpec extends ControllerBaseSpec {
+
+  private implicit val mockPsrSubmissionService: PsrSubmissionService = mock[PsrSubmissionService]
+
+  override protected val additionalBindings: List[GuiceableModule] =
+    List(bind[PsrSubmissionService].toInstance(mockPsrSubmissionService))
+
+  override protected def beforeEach(): Unit = {
+    reset(mockPsrSubmissionService)
+    when(mockPsrSubmissionService.submitPsrDetails(any(), any())(any(), any(), any()))
+      .thenReturn(Future.successful(Some(())))
+  }
+
+  private def onPageLoad(mode: Mode) =
+    routes.SharesDisposalCYAController.onPageLoad(srn, shareIndex, disposalIndex, mode)
+  private def onSubmit(mode: Mode) =
+    routes.SharesDisposalCYAController.onSubmit(srn, shareIndex, disposalIndex, mode)
+
+  private val shareIndex = refineMV[OneTo5000](1)
+  private val disposalIndex = refineMV[OneTo50](1)
+
+  private val sharesType = SponsoringEmployer
+  private val acquisitionType = Acquisition
+  private val acquisitionDate = Some(localDate)
+
+  private val howSharesDisposed = Sold
+
+  private val dateSharesSold = Some(localDate)
+  private val numberSharesSold = Some(totalShares)
+  private val considerationSharesSold = Some(money)
+  private val buyerIdentity = Some(Individual)
+  private val nameOfBuyer = Some(buyerName)
+//  private val buyerDetails = Some(nino)
```

Comment:
Some comments in this file, also further down. Can be fixed in a separate branch / with next ticket.

Reply:
Yup, these will be resolved along with the rest of the loose ends in the next ticket.